dask/array/slicing.py

```diff
@@ -560,45 +548,40 @@ def take(outname, inname, blockdims, index, axis=0):
 
     When list is sorted we retain original block structure
 
-    >>> blockdims, dsk = take('y', 'x', [(20, 20, 20, 20)], [1, 3, 5, 47], axis=0)
-    >>> blockdims
+    >>> chunks, dsk = take('y', 'x', [(20, 20, 20, 20)], [1, 3, 5, 47], axis=0)
+    >>> chunks
     ((3, 1),)
     >>> dsk  # doctest: +SKIP
     {('y', 0): (getitem, ('x', 0), ([1, 3, 5],)),
      ('y', 2): (getitem, ('x', 2), ([7],))}
     """
+    plan = slicing_plan(chunks[axis], index)
+    if len(plan) >= len(chunks[axis]) * 10:
+        factor = math.ceil(len(plan) / len(chunks[axis]))
+        warnings.warn("Slicing with an out-of-order index is generating %d "
+                      "times more chunks" % factor)
```

Comment:
Two suggestions to make this a little more usable:

1. Make an `Warning` subclass (e.g., `dask.array.PerformanceWarning`) to issue here instead of the default runtime warning. We could use this in `unify_chunks`, too. Then at least you can write `warnings.simplefilter('ignore', PerformanceWarning)` to silence this.
2. Include `stacklevel` (if possible) to point the warning message to the relevant line of indexing code.

Reply:
Done.  I wasn't aware of `stacklevel` before.  That's really nice!